From 7c57542a900d1033ea091489277620d0c631fa57 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 27 Aug 2020 07:57:54 -0700
Subject: [PATCH 1538/1851] v4l: xilinx: vip: Fix V4L2_PIX_FMT_XY10 bpl_factor

commit 66c78b7172772e6bbffb98364f4f761cafc7270a from
https://github.com/Xilinx/linux-xlnx.git

Fix the bpl_factor for V4L2_PIX_FMT_XY10 format.

The XY10 format stores 3 x 10 bit Luma (Y) only pixels in 32 bits.
E.g. for a 1920x1080 luma only buffer, the line width would be
1920 pixels / 3 pixels per word * 4 bytes per word = 2560 bytes.

With bpl_factor as 4, the width incorrectly becomes
1920 * 4 * 32 * 10 / (30 * 8) = 10240 bytes

Hence the bpl_factor is changed from 4 to 1 for V4L2_PIX_FMT_XY10.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vip.c b/drivers/media/platform/xilinx/xilinx-vip.c
index a9559a518938..fdda2b0b8120 100644
--- a/drivers/media/platform/xilinx/xilinx-vip.c
+++ b/drivers/media/platform/xilinx/xilinx-vip.c
@@ -91,7 +91,7 @@ static const struct xvip_video_format xvip_video_formats[] = {
 	{ XVIP_VF_MONO_SENSOR, 8, "mono", MEDIA_BUS_FMT_Y8_1X8,
 	  1, 8, V4L2_PIX_FMT_GREY, 1, 1, 1, 1 },
 	{ XVIP_VF_Y_GREY, 10, NULL, MEDIA_BUS_FMT_Y10_1X10,
-	  4, 32, V4L2_PIX_FMT_XY10, 1, 1, 1, 1 },
+	  1, 32, V4L2_PIX_FMT_XY10, 1, 1, 1, 1 },
 	{ XVIP_VF_Y_GREY, 12, NULL, MEDIA_BUS_FMT_Y12_1X12,
 	  1, 12, V4L2_PIX_FMT_XY12, 1, 1, 1, 1 },
 	{ XVIP_VF_Y_GREY, 16, NULL, MEDIA_BUS_FMT_Y16_1X16,
-- 
2.31.1

